README
NAME
    PYX::Utils - A perl module for PYX common utilities.

SYNOPSIS
     use PYX::Utils;

     my $decoded_text = decode($text);
     my $encoded_text = encode($text);
     my $decoded_text = entity_decode($text);
     my $encoded_text = entity_encode($text);

SUBROUTINES
  "decode"
     my $decoded_text = decode($text);

    Decode characters. Currently decode newline to '\n'.

    Returns decoded text.

  "encode"
     my $encoded_text = encode($text);

    Encode characters. Currently encode '\n' to newline.

    Returns encoded text.

  "entity_decode"
     my $decoded_text = entity_decode($text);

    Decode entities.

     '&lt;' => '<'
     '&amp;' => '&'
     '&quot;' => '"'

    Returns decoded text.

  "entity_encode"
     my $encoded_text = entity_encode($text);

    Encode some chars for HTML/XML/SGML. Currenctly encode these characters:

     '<' => '&lt;'
     '&' => '&amp;'
     '"' => '&quot;'

    Returns encoded text.

EXAMPLE1
     use strict;
     use warnings;

     use PYX::Utils qw(decode);

     # Text.
     my $text = "foo\nbar";

     # Decode.
     my $decoded_text = decode($text);

     # Print to output.
     print "Text: $text\n";
     print "Decoded text: $decoded_text\n";

     # Output:
     # Text: foo
     # bar
     # Decoded text: foo\nbar

EXAMPLE2
     use strict;
     use warnings;

     use PYX::Utils qw(encode);

     # Text.
     my $text = 'foo\nbar';

     # Encode text.
     my $encoded_text = encode($text);

     # Print to output.
     print "Text: $text\n";
     print "Encoded text: $encoded_text\n";

     # Output:
     # Text: foo\nbar
     # Encoded text: foo
     # bar

EXAMPLE3
     use strict;
     use warnings;

     use PYX::Utils qw(entity_decode);

     # Text.
     my $text = 'foo&lt;&amp;&quot;bar';

     # Decode entities.
     my $decoded_text = entity_decode($text);

     # Print to output.
     print "Text: $text\n";
     print "Decoded entities: $decoded_text\n";

     # Output:
     # Text: foo&lt;&amp;&quot;bar
     # Decoded entities: foo<&"bar

EXAMPLE4
     use strict;
     use warnings;

     use PYX::Utils qw(entity_encode);

     # Text.
     my $text = 'foo<&"bar';

     # Encode entities.
     my $encoded_text = entity_encode($text);

     # Print to output.
     print "Text: $text\n";
     print "Encoded text: $encoded_text\n";

     # Output:
     # Text: foo<&"bar
     # Encoded text: foo&lt;&amp;&quot;bar

DEPENDENCIES
    Exporter, HTML::Entities, Readonly.

SEE ALSO
    Task::PYX
        Install the PYX modules.

REPOSITORY
    <https://github.com/michal-josef-spacek/PYX-Utils>

AUTHOR
    Michal Josef Špaček skim@cpan.org

LICENSE AND COPYRIGHT
    © 2005-2021 Michal Josef Špaček

    BSD 2-Clause License

VERSION
    0.08



Changes
0.08

0.07 2021-01-26T21:32:59+01:00
 - Fix bugtracker.
 - Rewrite doc to use =head2 in SUBROUTINES section.
 - Update copyright years.

0.06 2020-03-23T11:18:38+01:00
 - Update author github username.

0.05 2020-03-23T11:09:26+01:00
 - Fix Makefile.PL fails with no '.' in @INC.
 - Fix SYNOPSIS section in doc.
 - Improve LICENSE AND COPYRIGHT section in doc.
 - Update Module::Install to 1.19 version.
 - Update author name.
 - Update copyright years.

0.04 2015-09-27T22:58:39+02:00
 - Improve SEE ALSO section in doc.
 - Move author tests to xt/ directory.
 - Update Module::Install to 1.16 version.

0.03 2015-02-02T07:34:43+01:00
 - Add REPOSITORY section to doc.
 - Add minimal version of Perl.
 - Fix SEE ALSO section to reality.
 - Update LICENSE AND COPYRIGHT section in doc to reality.
 - Update copyright years.

0.02 2014-11-19T09:35:41+01:00
 - Fix dependencies.

0.01 2014-11-17T23:00:14+01:00
 - First version.


Utils.pm
package PYX::Utils;

use base qw(Exporter);
use strict;
use warnings;

use HTML::Entities qw(decode_entities);
use Readonly;

# Constants.
Readonly::Array our @EXPORT_OK => qw(decode encode entity_decode entity_encode);
Readonly::Hash our %ENTITIES => (
        '<' => '&lt;',
        q{&} => '&amp;',
        q{"} => '&quot;',
);
Readonly::Scalar our $ENTITIES => join q{}, keys %ENTITIES;

our $VERSION = 0.08;

# Decode chars.
sub decode {
	my $text = shift;
	$text =~ s/\n/\\n/gms;
	return $text;
}

# Encode chars.
sub encode {
	my $text = shift;
	$text =~ s/\\n/\n/gms;
	return $text;
}

# Decode entities.
sub entity_decode {
	my $text = shift;
	return decode_entities($text);
}

# Encode some chars for HTML/XML/SGML.
sub entity_encode {
	my $text = shift;
	$text =~ s/([$ENTITIES])/$ENTITIES{$1}/gms;
	return $text;
}

1;

__END__

=pod

=encoding utf8

=head1 NAME

PYX::Utils - A perl module for PYX common utilities.

=head1 SYNOPSIS

 use PYX::Utils;

 my $decoded_text = decode($text);
 my $encoded_text = encode($text);
 my $decoded_text = entity_decode($text);
 my $encoded_text = entity_encode($text);

=head1 SUBROUTINES

=head2 C<decode>

 my $decoded_text = decode($text);

Decode characters.
Currently decode newline to '\n'.

Returns decoded text.

=head2 C<encode>

 my $encoded_text = encode($text);

Encode characters.
Currently encode '\n' to newline.

Returns encoded text.

=head2 C<entity_decode>

 my $decoded_text = entity_decode($text);

Decode entities.

 '&lt;' => '<'
 '&amp;' => '&'
 '&quot;' => '"'

Returns decoded text.

=head2 C<entity_encode>

 my $encoded_text = entity_encode($text);

Encode some chars for HTML/XML/SGML.
Currenctly encode these characters:

 '<' => '&lt;'
 '&' => '&amp;'
 '"' => '&quot;'

Returns encoded text.

=head1 EXAMPLE1

 use strict;
 use warnings;

 use PYX::Utils qw(decode);

 # Text.
 my $text = "foo\nbar";

 # Decode.
 my $decoded_text = decode($text);

 # Print to output.
 print "Text: $text\n";
 print "Decoded text: $decoded_text\n";

 # Output:
 # Text: foo
 # bar
 # Decoded text: foo\nbar

=head1 EXAMPLE2

 use strict;
 use warnings;

 use PYX::Utils qw(encode);

 # Text.
 my $text = 'foo\nbar';

 # Encode text.
 my $encoded_text = encode($text);

 # Print to output.
 print "Text: $text\n";
 print "Encoded text: $encoded_text\n";

 # Output:
 # Text: foo\nbar
 # Encoded text: foo
 # bar

=head1 EXAMPLE3

 use strict;
 use warnings;

 use PYX::Utils qw(entity_decode);

 # Text.
 my $text = 'foo&lt;&amp;&quot;bar';

 # Decode entities.
 my $decoded_text = entity_decode($text);

 # Print to output.
 print "Text: $text\n";
 print "Decoded entities: $decoded_text\n";

 # Output:
 # Text: foo&lt;&amp;&quot;bar
 # Decoded entities: foo<&"bar

=head1 EXAMPLE4

 use strict;
 use warnings;

 use PYX::Utils qw(entity_encode);

 # Text.
 my $text = 'foo<&"bar';

 # Encode entities.
 my $encoded_text = entity_encode($text);

 # Print to output.
 print "Text: $text\n";
 print "Encoded text: $encoded_text\n";

 # Output:
 # Text: foo<&"bar
 # Encoded text: foo&lt;&amp;&quot;bar

=head1 DEPENDENCIES

L<Exporter>,
L<HTML::Entities>,
L<Readonly>.

=head1 SEE ALSO

=over

=item L<Task::PYX>

Install the PYX modules.

=back

=head1 REPOSITORY

L<https://github.com/michal-josef-spacek/PYX-Utils>

=head1 AUTHOR

Michal Josef Špaček L<skim@cpan.org>

=head1 LICENSE AND COPYRIGHT

© 2005-2021 Michal Josef Špaček

BSD 2-Clause License

=head1 VERSION

0.08

=cut


Makefile.PL
use lib '.';
use strict;
use warnings;

use inc::Module::Install;

# Definition.
abstract 'A perl module for PYX common utilities';
author 'Michal Josef Spacek <skim@cpan.org>';
author_requires 'English' => 0;
author_requires 'File::Object' => 0.07;
author_requires 'Test::More' => 0;
author_requires 'Test::NoWarnings' => 0;
author_requires 'Test::Pod' => 0;
author_requires 'Test::Pod::Coverage' => 0;
license 'bsd';
name 'PYX-Utils';
readme_from 'Utils.pm';
recursive_author_tests('xt');
requires 'Exporter' => 0;
requires 'HTML::Entities' => 0;
requires 'Readonly' => 0;
requires 'perl' => '5.8.0';
resources 'bugtracker' => 'https://github.com/michal-josef-spacek/PYX-Utils/issues';
resources 'homepage' => 'https://github.com/michal-josef-spacek/PYX-Utils';
resources 'repository' => 'git://github.com/michal-josef-spacek/PYX-Utils';
test_requires 'Test::More' => 0;
test_requires 'Test::NoWarnings' => 0;
tests_recursive;
version '0.08';

# Run.
WriteAll();
